### DISCUSSION TITLE
Analytics for concourse

### DISCUSSION THREAD

**Discussion by VAIBHAVCES** [concourse/concourse#8600] [Ideas]
Concourse pipelines have a nice UI to see the latest runs and easily debug the problem, 

- I am using concourse for our business logic and I very strongly feel a lack of analytics here

- We have 6-7 pipelines running at different schedules and we have our on-call engineers monitoring these pipelines

- But the problem is some of them are very critical pipelines and some of them are less critical but at the end of the week both of them are important

- We need manual developer efforts to document these pipelines which creates developer fatigue

**Major Problem in summary**

1.  Lack of analytics
2.  Static report able to detect patterns among error messages

**Proposal**

1. Very easy and straightforward solution for users is to implement scripts to push data to their DBS and create their own report
2. Another way is at the concourse end we can design integrated analytics that can hold success and failure rates

Please share your suggestions on what solutions you have implemented, 
for now, one temporary solution I am using is implementing a python script that pushes pipeline run status to a confluence, and then I have created an analytics report from that data  
In my opinion, this is not a very scalable solution hence looking forward for an integration in concourse or a better scalable solution

**Comment by suhlig** [comment]
We handle this with a custom resource that emits metrics about what the pipeline achieved (or failed to do so), and then alert on these metrics (totally outside Concourse). Works well for us.

We also looked at [Concourse' own metrics](https://concourse-ci.org/metrics.html), but found them to be too broad for our needs.
